### PR TITLE
implement tealv3 byte opcodes

### PR DESCRIPTION
**Pull request by amityadav0**
- implement tealv3 byte opcodes
- add tests

### CODE REVIEW

File: packages/runtime/src/interpreter/opcode-list.ts [hunk 1]
```diff
@@ -2368,3 +2368,63 @@ export class GetBit extends Op {
...
+
+    target[Number(index)] = Number(smallInteger);
+    stack.push(target);
+  }
+}
+
+/**
+ * pop a byte-array A and integer B. Extract the Bth byte of A and push it as an integer
+ * Pops from stack: ... stack, {[]byte A}, {uint64 B}
+ * Pushes to stack: uint64
```
Comment (ratik21) [inline packages/runtime/src/interpreter/opcode-list.ts:2407]:
```suggestion
 * Pops from stack: [ ...stack, {[]byte A}, {uint64 B} ]
 * Pushes to stack: [ ...stack, uint64 ]
```

File: packages/runtime/test/src/interpreter/opcode-list.ts [hunk 2]
```diff
@@ -3607,4 +3608,147 @@ describe("Teal Opcodes", function () {
...
+      stack.push(new Uint8Array(0)); // target
+      stack.push(new Uint8Array(0)); // index
+
+      expectRuntimeError(
+        () => op.execute(stack),
+        RUNTIME_ERRORS.TEAL.INVALID_TYPE
+      );
+    });
+
+    it("should panic if index bit in out of bytes array", () => {
```
Comment (ratik21) [inline packages/runtime/test/src/interpreter/opcode-list.ts:3659]: could you please check if this line is correct?

File: packages/runtime/test/src/interpreter/opcode-list.ts [hunk 3]
```diff
@@ -3607,4 +3608,147 @@ describe("Teal Opcodes", function () {
...
+      stack.push(10n); // target
+      stack.push(0n); // index
+
+      expectRuntimeError(
+        () => op.execute(stack),
+        RUNTIME_ERRORS.TEAL.INVALID_TYPE
+      );
+    });
+
+    it("should panic if index is not uint", () => {
```
Comment (ratik21) [inline packages/runtime/test/src/interpreter/opcode-list.ts:3648]: `unit64` or `uint8`?

File: packages/runtime/test/src/interpreter/opcode-list.ts [hunk 4]
```diff
@@ -3607,4 +3608,147 @@ describe("Teal Opcodes", function () {
...
+
+      stack.push(new Uint8Array([8, 2, 1, 9])); // target
+      stack.push(3n); // index
+      stack.push(0n); // small integer
+
+      op.execute(stack);
+      assert.deepEqual(stack.pop(), new Uint8Array([8, 2, 1, 0]));
+    });
+
+    it("should panic if target is not bytes", () => {
```
Comment (ratik21) [inline packages/runtime/test/src/interpreter/opcode-list.ts:3700]:
```suggestion
    it("should panic if target is not bytes(Uint8Array)", () => {
```

File: packages/runtime/test/src/interpreter/opcode-list.ts [hunk 5]
```diff
@@ -3607,4 +3608,147 @@ describe("Teal Opcodes", function () {
...
+        RUNTIME_ERRORS.TEAL.SET_BIT_INDEX_BYTES_ERROR
+      );
+    });
+  });
+
+  describe("SetByte", () => {
+    let stack: Stack<StackElem>;
+    this.beforeEach(() => { stack = new Stack<StackElem>(); });
+
+    it("should set correct bytes in stack", () => {
```
Comment (ratik21) [inline packages/runtime/test/src/interpreter/opcode-list.ts:3683]:
```suggestion
    it("should set correct bytes and push to stack", () => {
```

File: packages/runtime/test/src/interpreter/opcode-list.ts [hunk 6]
```diff
@@ -3607,4 +3608,147 @@ describe("Teal Opcodes", function () {
...
+      stack.push(1n); // index
+      stack.push(new Uint8Array(0));
+
+      expectRuntimeError(
+        () => op.execute(stack),
+        RUNTIME_ERRORS.TEAL.INVALID_TYPE
+      );
+    });
+
+    it("should panic if index bit in out of bytes array", () => {
```
Comment (ratik21) [inline packages/runtime/test/src/interpreter/opcode-list.ts:3733]: could you please check if this line is correct?

**Review by ratik21 — APPROVED** [review]
left few suggestions

File: packages/runtime/test/src/interpreter/opcode-list.ts (hunk 3 above)
Comment (amityadav0) [reply packages/runtime/test/src/interpreter/opcode-list.ts:3648]: this test is for byte cheking `index should not be bytes`

File: packages/runtime/test/src/interpreter/opcode-list.ts (hunk 2 above)
Comment (amityadav0) [reply packages/runtime/test/src/interpreter/opcode-list.ts:3659]: ok

File: packages/runtime/test/src/interpreter/opcode-list.ts (hunk 6 above)
Comment (amityadav0) [reply packages/runtime/test/src/interpreter/opcode-list.ts:3733]: ok

**Pull request merged**